enhancements/rhcos/rhcos-inject.md

```diff
@@ -0,0 +1,198 @@
+
+## Proposal
+
+Add a new, optional step to the installation process that allows an admin to inject network configuration and any other customization to the RHCOS installer image (and maybe other RHCOS images in the future) before invocation. This customization is performed by a new `rhcos-inject` utility that takes as input an RHCOS installer image (`rhcos-<version>-installer.<architecture>.iso`), network configuration, and a set of Ignition Configs and generates a new RHCOS installer image with all of the configuration and customization included. This new installer image can then be booted in an unsupervised manner and will complete the RHCOS installation along with any customization.
+
+Initially, `rhcos-inject` will primarily focus on network configuration but as the needs of customers evolve, this can be expanded to include other common customization. As always, an escape hatch is needed for unanticipated cases and this will be in the form of raw Ignition Configs. If there is a need for additional customization beyond network configuration, the admin can include that in an Ignition Config and inject that alongside everything else.
+
+An example of its invocation can be seen here:
+
+```console
+$ rhcos-inject \
+	--input=rhcos-installer.x86_64.iso \
+	--output=control-0.iso \
+	--openshift-config=master.ign \
+	--bond=bond0:em1,em2:mode=active-backup \
+	--ip=10.10.10.2::10.10.10.254:255.255.255.0:control0.example.com:bond0:none
+```
+
+With a little scripting, a number of custom RHCOS installers can be quickly created:
+
+```zsh
+for i in {00..11}
+do
+	rhcos-inject \
+		--input=rhcos-installer.x86_64.iso \
+		--output=worker-${i}.iso \
+		--openshift-config=worker.ign \
+		--ip=10.10.10.$((i+10))::10.10.10.254:255.255.255.0:worker${i}.example.com:enp1s0:none
+done
+```
+
+This network configuration applies to the post-pivot RHCOS installation environment and is copied to both the pre- and post-pivot installed environment by default. In the event that exotic configuration is required, this copying of configuration can be disabled and a service can be used instead (TODO(crawford) figure out systemd service ordering):
+
+```console
+$ cat rhcp.ign
+{
+  "ignition": { "version": "3.0.0"},
+  "systemd": {
+    "units": [{
+      "name": "random-host-configuration-protocol.service",
+      "enabled": true,
+      "contents": "[Service]\nType=oneshot\nExecStart=/usr/bin/env nmcli ...\n\n[Install]\nWantedBy=pre-install.target"
+    }]
+  }
+}
+
+$ rhcos-inject \
+	--input=rhcos-installer.x86_64.iso \
+	--output=worker.iso \
+	--openshift-config=worker.ign \
+	--installer-config=rhcp.ign \
+	--persist-networking=false
+```
+
+This custom service runs in the context of the RHCOS installer and is responsible for writing the network configuration for the installed system (e.g. based on an external lookup by MAC address). This escape hatch can be used in environments which employ an in-house host configuration procedure.
```

Comment:
We should probably provide some sugar for generating the [install hook boilerplate](https://github.com/coreos/coreos-installer/blob/master/docs/customizing-install.md#hooking-coreos-installer-at-boot-time).